azkaban-hadoop-security-plugin/src/main/java/azkaban/security/HadoopSecurityManager_H_2_0.java

```diff
@@ -348,8 +358,9 @@ private void registerCustomCredential(final Props props, final Credentials hadoo
   public KeyStore getKeyStore(final Props props) {
     logger.info("Prefetching KeyStore for the flow");
     final Credentials cred = new Credentials();
-    final CredentialProviderWithKeyStore customCredential = getCustomCredentialProvider(
-            props, cred, logger, Constants.ConfigurationKeys.CUSTOM_CREDENTIAL_NAME);
+    final CredentialProviderWithKeyStore customCredential = (CredentialProviderWithKeyStore)
```

Comment:
No need to guard this cast because getKeyStore() should never be called with non-keystore provider, right? Probably fine... but will bad things happen if that assumption is violated? Again, probably fine.

Reply:
I had it initially like that, it wont even compile without the cast.